statement_test.go

```diff
@@ -844,3 +844,21 @@ func TestDriverValuer(t *testing.T) {
 	require.Error(t, err, "[]uuid.UUID should fail without driver.Valuer")
 	require.Contains(t, err.Error(), "unsupported data type: UUID")
 }
+
+func TestMixedTypeSliceBinding(t *testing.T) {
+	db := openDbWrapper(t, ``)
+	defer closeDbWrapper(t, db)
+
+	createTable(t, db, `CREATE TABLE mixed_slice_test (foo TEXT)`)
+	_, err := db.Exec(`INSERT INTO mixed_slice_test VALUES ('hello this is text'), ('other')`)
+	require.NoError(t, err)
```

Comment:
Let's also add a test for the same-type case (with `any`)? E.g., `[]any{"hello this is text", "other"}`.

Reply:
Done